WeScan/Common/VisionRectangleDetector.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  VisionRectangleDetector.swift
+//  WeScan
+//
+//  Created by Julian Schiavo on 28/7/2018.
+//  Copyright © 2018 WeTransfer. All rights reserved.
+//
+
+import Vision
+import Foundation
+
+/// Class used to detect rectangles from an image.
+@available(iOS 11.0, *)
+struct VisionRectangleDetector {
+    
+    /// Detects rectangles from the given image on iOS 11 and above.
+    ///
+    /// - Parameters:
+    ///   - image: The image to detect rectangles on.
+    /// - Returns: The biggest detected rectangle on the image.
+    static func rectangle(forImage image: CIImage, completion: @escaping ((Quadrilateral?) -> Void)) {
+      
+        let imageRequestHandler = VNImageRequestHandler(ciImage: image, options: [:])
+
+        // Create the rectangle request, and, if found, return the biggest rectangle (else return nothing).
+        let rectangleDetectionRequest: VNDetectRectanglesRequest = {
+            let rectDetectRequest = VNDetectRectanglesRequest(completionHandler: { (request, error) in
+                if error == nil {
```

Comment:
We should use a guard here:
``` guard error == nil else {
    ...
}
```

Reply:
This has been implemented, and the guard statement has been updated to check for an empty results array so the current rectangle can be removed.